src/cpp/benders/benders_mpi/BendersMPI.cpp

```diff
@@ -30,22 +30,41 @@ BendersMpi::BendersMpi(const BendersBaseOptions& options,
 void BendersMpi::InitializeProblems()
 {
     MatchProblemToId();
-
     BuildMasterProblem();
-    int current_problem_id = 0;
-    // Dispatch subproblems to process
-    for (const auto& problem: coupling_map_)
+    if (_options.CACHE_PROBLEMS)
     {
-        // In case there are more subproblems than process
-        if (auto process_to_feed = current_problem_id % _world.size();
-            process_to_feed == _world.rank())
-        { // Assign  [problemNumber % processCount] to processID
-
-            const auto subProblemFilePath = GetSubproblemPath(problem.first);
-            AddSubproblem(problem);
-            AddSubproblemName(problem.first);
+        int current_problem_id = 0;
```

Comment:
What happens here in the if part, I do not undertand the purpose of this code block ?

Reply:
Ah you pass a coupling map with only the needed problems to each process !